### PR TITLE
chore: upgrade json schema to support grouping by platform MONGOSH-1346

### PR DESCRIPTION
**Pull request by kmruiz**

### CODE REVIEW

File: src/download-center-config.ts [hunk 1]
```diff
@@ -19,3 +21,24 @@ export interface DownloadCenterConfig {
   supported_browsers_link: string;
   tutorial_link: string;
 }
+export interface DownloadCenterConfigV2 {
+  previous_releases_link: string;
+  release_notes_link: string;
+  tutorial_link: string;
+  version: string;
```
Comment (addaleax) [inline src/download-center-config.ts]: This feels odd – are multiple versions not supported in this variant?

Are they really fully distinct download center config schemas? I would have expected that the basic structure of the download center config would have stayed the same here, and we would just have applied the union type to `platform`, for example 

```js
    platform: ({
      arch: string;
      os: string;
      name: string;
    } & ({
      download_link: string;
    } | {
      packages: Package[];
    }))[];
```

Comment (kmruiz) [reply src/download-center-config.ts]: I agree with you, but I wrote the schema based on  https://github.com/mongodb/mongodb-atlas-cli/blob/main/tools/release/main.go#L58 which only supports one version. I'm still waiting for an answer from the Web Team for a formal json schema (or something) that I can use as an example.

Comment (kmruiz) [reply src/download-center-config.ts]: I've talked to @mcasimir and ~he~ we found a solution that should work as you are mentioning. I've pushed the code with the changes, when you feel better (*no rush*) take a look.

**Comment by kmruiz** [comment]
Paired with @mcasimir

**Review by mcasimir — APPROVED** [review]

**Pull request merged**